### PR TITLE
Create cpp Solution and Analysis to USACO Snow Cow

### PR DESCRIPTION
**Pull request by dutinmeow**
Some of the formatting might be messed up because I'm not too familiar with markdown.

### CODE REVIEW

File: solutions/USACO feb19-snowcow.mdx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_snowcow_platinum_dec19.html)
+
+## Explanation
+
+Although the official solution uses manipulation on two BITs, its more straightforward to use lazy propogation on a segment tree. 
+As with the official solution, we first create an array using a preorder traversal of the tree: thus, each node's subtree lies in
+a contiguous range following the position of that node. We store the start and end points of the range of each node x's subtree in 
+arrays \texttt{LL[x]} and \texttt{RR[x]}. Note that each node's subtree size can be calculated by simply subtracting the bounds of this range.
```
Comment (maggie-j-liu) [inline solutions/USACO feb19-snowcow.mdx:15]:
```suggestion
arrays $\texttt{LL[x]}$ and $\texttt{RR[x]}$. Note that each node's subtree size can be calculated by simply subtracting the bounds of this range.
```

File: solutions/USACO feb19-snowcow.mdx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+[Official Analysis](http://www.usaco.org/current/data/sol_snowcow_platinum_dec19.html)
+
+## Explanation
+
+Although the official solution uses manipulation on two BITs, its more straightforward to use lazy propogation on a segment tree. 
+As with the official solution, we first create an array using a preorder traversal of the tree: thus, each node's subtree lies in
+a contiguous range following the position of that node. We store the start and end points of the range of each node x's subtree in 
+arrays $\texttt{LL[x]}$ and $\texttt{RR[x]}$. Note that each node's subtree size can be calculated by simply subtracting the bounds of this range.
+
+We then construct a lazy segment tree on an empty array as a base. This will represent the number of distinct colors at each node
```
Comment (dongliu0426) [inline solutions/USACO feb19-snowcow.mdx:17]: add a period

File: solutions/USACO feb19-snowcow.mdx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+Note that each query can be calculated by querying the sum of distinct colors from the endpoints of each node. 
+Similarly to the official solution, we keep track of updates for each color. For each new update ($c$, $x$), we only update the tree by:
```
Comment (dongliu0426) [inline solutions/USACO feb19-snowcow.mdx:19]:
```suggestion
Similarly to the official solution, we keep track of updates for each color. For each new update $(c, x)$, we only update the tree by: 
```

**Review by dongliu0426 — CHANGES_REQUESTED** [review]
code looks good

**Review by dongliu0426 — APPROVED** [review]

**Comment by dolphingarlic** [comment]
Using two BITs is essentially the same as using a lazy segtree for range update range sum though

Also there's no need to repeat what's already in the editorial

**Comment by dolphingarlic** [comment]
@dutinmeow you still there?

**Comment by dolphingarlic** [comment]
I guess not...

Feel free to reopen if you come back to this

**Pull request closed without merging**